### ISSUE TITLE
Error with malli 0.20.1 when using malli.experimental.time

### ISSUE THREAD

**Issue by skynet-gh** [NoahTheDuke/splint#40]
With [malli 0.20.1](https://github.com/metosin/malli/releases/tag/0.20.1) there's an error with a `.cljc` file like this:
```
(ns ...
  (:require
    #?@(:clj
        [[malli.core :as m]
         [malli.experimental.time :as met]
         [malli.experimental.time.generator]
         [malli.registry :as mr]])
    ))

#?(:clj
   (mr/set-default-registry!
     (mr/composite-registry
       (m/default-schemas)
       (met/schemas))))
```

(for [java.time schemas](https://github.com/metosin/malli?tab=readme-ov-file#malliexperimentaltime))
it fails with this message:

```
$ clojure -M:splint
<filename>:null:null [splint/unknown-error] - Splint encountered an error: class java.lang.Object cannot be cast to class clojure.lang.IObj (java.lang.Object is in module java.base of loader 'bootstrap'; clojure.lang.IObj is in unnamed module of loader 'app')
```

otherwise, when I remove these code bits, it works normally on the whole project.

**Comment by NoahTheDuke** [comment]
Thank you for the report!

**Comment by NoahTheDuke** [comment]
I tried to reproduce this and cannot. Can you give me more details?

**Comment by skynet-gh** [comment]
@NoahTheDuke given:
`deps.edn`:
```clj
{:paths ["src"]
 :deps
 {metosin/malli {:mvn/version "0.20.1"}}
 :aliases
 {:splint
  {:extra-deps {io.github.noahtheduke/splint {:mvn/version "1.23.1"}}
   :main-opts ["-m" "noahtheduke.splint"]}}}
```
`src/splint/bug.cljc`:
```clj
(ns splint.bug
  (:require
    clojure.string
    #?@(:clj [[malli.core :as m]
              [malli.experimental.time :as met]
              [malli.experimental.time.generator]
              [malli.registry :as mr]])))

#?(:clj
   (mr/set-default-registry!
     (mr/composite-registry
       (m/default-schemas)
       (met/schemas))))
```
then run:
```sh
$ clojure -M:splint
src/splint/bug.cljc:null:null [splint/unknown-error] - Splint encountered an error: class java.lang.Object cannot be cast to class clojure.lang.IObj (java.lang.Object is in module java.base of loader 'bootstrap'; clojure.lang.IObj is in unnamed module of loader 'app')

Linting took 38ms, checked 1 files, 0 style warnings, 1 errors
```
but in the process of setting this up, I noticed that adding a case to the reader conditional for `:cljs` fixes it:
```clj
#?@(:clj ... :cljs [])
```
```sh
$ clojure -M:splint
Linting took 37ms, checked 1 files, 0 style warnings
```

**Comment by NoahTheDuke** [comment]
Okay, I figured it out. This comes from `malli` 0.20.1 bringing in a newer version of `borkdude/edamame` (clojure code parser) than anticipated, which has a slight change in how it works. I'll have to see if I can adjust my code to account for this, or otherwise fix it in edamame itself.

To avoid the problem right now, you can add `borkdude/edamame {:mvn/version "1.5.35"}` to your splint alias, which should fix the issue.

**Comment by NoahTheDuke** [comment]
Tracking the fix in https://github.com/borkdude/edamame/issues/141

**Comment by NoahTheDuke** [comment]
Fixed and released in 1.24.0.
